upgrade/scripts/upgrade_python_package.py

```diff
@@ -216,7 +220,12 @@ def is_cloudsmith_url_valid(cloudsmith_url):
 
 def is_package_already_installed(package):
     results = pip("list", "--format", "json")
-    parsed_results = json.loads(results)
+    try:
+        decoder = json.JSONDecoder()
+        parsed_results, index = decoder.raw_decode(results)
+    except json.JSONDecodeError as e:
+        logging.error(f"Error occurred while decoding pip list at ${index} index.")
+        raise e
```

Comment:
Do we want the script to fail if this fails? Maybe we do in order to avoid any chance of having broken dependencies. However, in that case we might want to ask the user to upgrade pip. So this can be a custom exception.

Reply:
We probably want to fail if pip --format json isn't parsed into a json correctly. That would IMO indicate that a deprecated (or potentially newer, breaking) pip is used? I'll add a custom exception

Reply:
The question is if we want to support really old pio too, or make sure that, say, version 18 is new enough.